src/v0/bitcoin/mod.rs

```diff
@@ -442,11 +444,14 @@ impl Psbt {
                     let (msg, sighash_type) = self.sighash_taproot(input_index, cache, None)?;
                     let key_pair = Keypair::from_secret_key(secp, &sk.inner)
                         .tap_tweak(secp, input.tap_merkle_root)
-                        .to_inner();
-
-                    #[cfg(feature = "rand")]
-                    let signature = secp.sign_schnorr(&msg, &key_pair);
-                    #[cfg(not(feature = "rand"))]
+                        .to_keypair();
+
+                    #[cfg(all(feature = "rand", feature = "std"))]
+                    let signature = {
+                        let mut rng = thread_rng();
+                        secp.sign_schnorr_with_rng(&msg, &key_pair, &mut rng)
+                    };
```

Comment:
Why this change please?

Reply:
Ah, i think I was getting tripped up on not having [`rand-std` enabled](https://docs.rs/secp256k1/0.29.1/secp256k1/struct.Secp256k1.html#method.sign_schnorr) or something. Let me see where I got confused.